RFCs/0043-Fn::FindInMap enhancements (default value and intrinsic functions).md

```diff
@@ -0,0 +1,571 @@
+
+An example is as follows. 
+
+### YAML
+```yaml
+Mappings:
+  DNS:
+    us:
+      dns: mypage.us.com
+      ttl: '600'
+    ca:
+      dns: mypage.us.com
+      ttl: '600'
+    mx:
+      dns: mypage.default.com
+      ttl: '300'
+    ar:
+      dns: mypage.default.com
+      ttl: '300'
+    br:
+      dns: mypage.default.com
+      ttl: '300'
+
+!FindInMap [DNS, !Ref country, dns]
+```
+
+### JSON
+```json
+{
+    "Mappings" : {
+        "DNS": {
+            "us" : {"dns" : "mypage.us.com", "ttl" : "600"},
+            "ca" : {"dns" : "mypage.us.com", "ttl" : "600"},
+            "mx" : {"dns" : "mypage.default.com", "ttl" : "300"},
+            "ar" : {"dns" : "mypage.default.com", "ttl" : "300"},
+            "br" : {"dns" : "mypage.default.com", "ttl" : "300"}
+        }
+    }
+}
+
+{ "Fn::FindInMap" : [ "DNS", { "Ref" : "country" }, "dns" ]}
+```
+
+The desired behavior would be something like this:
+
+```yaml
+!If
+- !MappingPresent [Mapping, 'foo', 'bar']
+- !FindInMap [Mapping, 'foo', 'bar']
+- 'MyDefaultValue'
+```
+
+The above example could then be simplified as follows.
+
+### YAML
+```yaml
+Mappings:
+  DNS:
+    us:
+      dns: mypage.us.com
+      ttl: '600'
+    ca:
+      dns: mypage.us.com
+      ttl: '600'
+
+!FindInMap [DNS, !Ref country, dns, mypage.default.com]
+```
+
+### JSON
+```json
+{
+    "Mappings" : {
+        "DNS": {
+            "us" : {"dns" : "mypage.us.com", "ttl" : "600"},
+            "ca" : {"dns" : "mypage.us.com", "ttl" : "600"}
+        }
+    }
+}
+
+{ "Fn::FindInMap" : [ "DNS", { "Ref" : "country" }, "dns", "mypage.default.com"]}
+```
+
+A current workaround can be found here (https://stackoverflow.com/questions/35904204/default-value-on-mappings-aws-cloudformation).
+
+This RFC seeks to create an easier built-in way for customers to specify a default value if a Mapping cannot be found. 
+
+* Reference:
+    * https://github.com/aws-cloudformation/cfn-language-discussion/issues/43
+    * https://stackoverflow.com/questions/35904204/default-value-on-mappings-aws-cloudformation
+
+# Examples
+
+## Use of other intrinsic functions:
+
+Using Fn::Select in Fn::FindInMap. 
+
+### YAML
+```yaml
+Parameters:
+
+  AsymmetricRSAKeyUsage:
+    Type: String
+    AllowedValues:
+      - ENCRYPT_DECRYPT
+      - SIGN_VERIFY
+    Default: ENCRYPT_DECRYPT
+
+  KeySpec:
+    Type: String
+    AllowedValues:
+      - ECC_NIST_P256
+      - ECC_NIST_P384
+      - ECC_NIST_P521
+      - ECC_SECG_P256K1
+      - HMAC_224
+      - HMAC_256
+      - HMAC_384
+      - HMAC_512
+      - RSA_2048
+      - RSA_3072
+      - RSA_4096
+      - SYMMETRIC_DEFAULT
+    Default: SYMMETRIC_DEFAULT
+
+Conditions:
+  IsKeyAsymmetricRSA: !Equals ['RSA', !Select [0, !Split [_, !Ref KeySpec]]]
```

Comment:
Sidenote: `!StartsWith` or `!MatchRegex` would be useful here (I couldn't immediately find the related issue, but I seem to remember something existing)

Reply:
https://github.com/aws-cloudformation/cfn-language-discussion/issues/87